extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -5,294 +5,310 @@
-		ignoreFocusOut: true
-	}));
+		// 3. ask for object name for the binding
+		const objectName = await this.promptForObjectName(selectedBindingType);
 
-	if (!selectedBinding) {
-		return;
-	}
+		if (!objectName) {
+			return;
+		}
 
-	// 3. ask for object name for the binding
-	const objectName = await vscode.window.showInputBox({
-		prompt: selectedBinding.type === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
-		placeHolder: constants.placeHolderObject,
-		validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
-		ignoreFocusOut: true
-	});
+		// 4. ask for connection string setting name
+		let projectUri: vscode.Uri | undefined;
+		try {
+			projectUri = await azureFunctionsUtils.getAFProjectContainingFile(uri);
+		} catch (e) {
+			// continue even if there's no AF project found. The binding should still be able to be added as long as there was an azure function found in the file earlier
+		}
 
-	if (!objectName) {
-		return;
-	}
+		let connectionStringSettingName = await this.promptForConnectionStringSetting(projectUri);
+		if (!connectionStringSettingName) {
+			return;
+		}
 
-	// 4. ask for connection string setting name
-	let projectUri: vscode.Uri | undefined;
-	try {
-		projectUri = await azureFunctionsUtils.getAFProjectContainingFile(uri);
-	} catch (e) {
-		// continue even if there's no AF project found. The binding should still be able to be added as long as there was an azure function found in the file earlier
-	}
+		// 5. insert binding
+		try {
+			const result = await azureFunctionsService.addSqlBinding(selectedBindingType, uri.fsPath, azureFunctionName, objectName, connectionStringSettingName);
 
-	let connectionStringSettingName;
+			if (!result.success) {
+				void vscode.window.showErrorMessage(result.errorMessage);
+				TelemetryReporter.sendErrorEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding);
+				return;
+			}
 
-	// show the settings from project's local.settings.json if there's an AF functions project
-	if (projectUri) {
-		let settings;
-		try {
-			settings = await azureFunctionsUtils.getLocalSettingsJson(path.join(path.dirname(projectUri.fsPath!), constants.azureFunctionLocalSettingsFileName));
+			TelemetryReporter.createActionEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding)
+				.withAdditionalProperties({ bindingType: BindingType[selectedBindingType] })
+				.send();
 		} catch (e) {
 			void vscode.window.showErrorMessage(utils.getErrorMessage(e));
+			TelemetryReporter.sendErrorEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding);
 			return;
 		}
+	}
 
-		let existingSettings: (vscode.QuickPickItem)[] = [];
-		if (settings?.Values) {
-			existingSettings = Object.keys(settings.Values).map(setting => {
-				return {
-					label: setting
-				} as vscode.QuickPickItem;
-			});
-		}
+	public async promptForBindingType(): Promise<BindingType | undefined> {
```

Comment:
I would say you can move these prompts to azureFunctionUtils - so that we could leverage them in other places as well.